Lib/pathlib.py

```diff
@@ -715,6 +715,10 @@ def match(self, path_pattern, *, case_sensitive=None):
                 return False
         return True
 
+# Subclassing os.PathLike makes isinstance() checks slower,
+# which in turn makes Path construction slower. Register instead!
+os.PathLike.register(PurePath)
```

Comment:
Note that you can use `ABC.register` as a class decorator, but I think it would not look better for this case.